### PR TITLE
(MINOR) User, UserCbor and Uuid assertions removed from SDK

### PR DESCRIPTION
**Pull request by gpeacock**
## Changes in this pull request
Remove User, UserCbor and Uuid assertions from the public SDK
Use add_labeled_assertion instead.
## Checklist
- [ ] This PR represents a single feature, fix, or change.
- [ ] All applicable changes have been documented.
- [ ] Any `TO DO` items (or similar) have been entered as GitHub issues and the link to that issue has been included in a comment.

### CODE REVIEW

File: sdk/src/manifest.rs [hunk 1]
```diff
@@ -699,26 +699,23 @@ impl Manifest {
     ///
     /// ```
     /// # use c2pa::Result;
-    /// use c2pa::{
-    ///     assertions::User,
-    ///     create_signer,
-    ///     Manifest,
-    ///     SigningAlg,
-    /// };
+    /// use c2pa::{create_signer, Manifest, SigningAlg,};
```
Comment (scouten-adobe) [inline sdk/src/manifest.rs:702]: Extra comma here shouldn't be necesary.

**Review by scouten-adobe — CHANGES_REQUESTED** [review]
Looks good. I'd like to see something in the README (breaking changes since 0.14.0 to call attention to how anybody using the now-private interface should update their code.

**Review by dkozma — APPROVED** [review]
LGTM after fixing @scouten-adobe 's comment

File: sdk/src/manifest.rs (hunk 1 above)
Comment (gpeacock) [reply sdk/src/manifest.rs]: removed

**Review by scouten-adobe — COMMENTED** [review]
I think any developer relying on existing functionality could feel pretty frustrated upon taking this upgrade and not having an obvious guide to how to migrate.

I'll grant that an upgrading guide isn't particularly common, but I've seen some crates that do this (for example, https://github.com/rust-random/rand#versions) and I think it's a pattern worth adopting.

FWIW I do this in the XMP Toolkit crate: https://github.com/adobe/xmp-toolkit-rs#breaking-changes-in-0x-series

What you've added in the changelog says that there is something that is breaking, but isn't particularly helpful in guiding any developer affected by the change toward a resolution of their problem.

I won't give this a :+1: without a guide, but I'll remove my change request.

**Comment by gpeacock** [comment]
Updated readme and pulled in the latest changes. The example using the User assertion has been replaced the an an example of using a custom struct and add_labeled_assertion

**Pull request merged**